### PR TITLE
fix: export managedAccount module

### PR DESCRIPTION
**Pull request by mcstover**
- Export the new managedAccount module
- Add options object for passing variables and fetchPolicy to new promo mutations and promo campaign query
- Added guards for missing required variables and additional tests

### CODE REVIEW

**Comment by mcstover** [comment]
Just realized there's some other things missing with the new mutations I had created, such as the ability to pass in variables. I'll add that here next week.

**Comment by mcstover** [comment]
Ok this PR is ready for review.

**Review by eddieferrer — APPROVED** [review]

File: @kiva/kv-shop/src/basketCredits.ts [hunk 1]
```diff
@@ -51,9 +51,16 @@ export interface ApplyPromoCreditData {
...
+export async function applyPromoCredit(
+	apollo: ApolloClient<any>,
+	options: MutationOptions<any>,
+): Promise<boolean> {
+	if (!options?.variables?.creditType && !options?.variables?.redemptionCode) {
+		return Promise.resolve(false);
+	}
 	const data = await callShopMutation<ApplyPromoCreditData>(apollo, {
 		awaitRefetchQueries: true,
+		fetchPolicy: options?.fetchPolicy ?? 'network-only',
```
Comment (dyersituations) [inline @kiva/kv-shop/src/basketCredits.ts:63]: Do we want the default to be `network-only`? The typical default is `cache-first`.

File: @kiva/kv-shop/src/basketCredits.ts [hunk 2]
```diff
@@ -76,9 +84,16 @@ export interface RemovePromoCreditData {
...
+export async function removePromoCredit(
+	apollo: ApolloClient<any>,
+	options: MutationOptions<any>,
+): Promise<boolean> {
+	if (!options?.variables?.creditType && !options?.variables?.redemptionCode) {
+		return Promise.resolve(false);
+	}
 	const data = await callShopMutation<RemovePromoCreditData>(apollo, {
 		awaitRefetchQueries: true,
+		fetchPolicy: options?.fetchPolicy ?? 'network-only',
```
Comment (dyersituations) [inline @kiva/kv-shop/src/basketCredits.ts:96]: Same policy question.

File: @kiva/kv-shop/src/managedAccount.ts [hunk 3]
```diff
@@ -107,7 +120,7 @@ export async function getPromoFromBasket(apollo: ApolloClient<any>): Promise<Sho
 				}
 			}
 		`,
-		fetchPolicy: 'network-only',
+		fetchPolicy: options?.fetchPolicy ?? 'network-only',
```
Comment (dyersituations) [inline @kiva/kv-shop/src/managedAccount.ts:123]: Same policy question.

File: @kiva/kv-shop/src/basketCredits.ts (hunk 1 above)
Comment (mcstover) [reply @kiva/kv-shop/src/basketCredits.ts:63]: These should always be network only as they need to update the basket state on the server

File: @kiva/kv-shop/src/managedAccount.ts (hunk 3 above)
Comment (mcstover) [reply @kiva/kv-shop/src/managedAccount.ts:123]: Same.

**Review by dyersituations — APPROVED** [review]

**Pull request merged**